packages/serverpod/lib/src/database/extensions.dart

```diff
@@ -101,150 +86,275 @@ extension TableComparisons on TableDefinition {
+      if (key == null) {
+        mismatches.add(
+            'Extra foreign key "${otherKey.constraintName}" found in the target schema.');
+      }
+    }
+
+    return mismatches;
   }
 }
 
 /// Comparison methods for [ColumnDefinition].
 extension ColumnComparisons on ColumnDefinition {
-  /// Returns true if the column structure is identical to the [other]. Ignores
-  /// comparisons of Dart types and names.
-  bool like(ColumnDefinition other) {
-    if (other.dartType != null &&
-        dartType != null &&
-        other.dartType != dartType) {
-      return false;
-    }
-
-    return (other.isNullable == isNullable &&
-        other.columnType.like(columnType) &&
-        other.name == name &&
-        other.columnDefault == columnDefault);
+  /// Compares this column definition with [other], returning a list of mismatches.
+  /// Returns an empty list if the columns are identical.
+  List<String> like(ColumnDefinition other) {
+    List<String> mismatches = [];
+
+    if (name != other.name) {
+      mismatches.add('Name mismatch: Expected "$name", found "${other.name}".');
+    }
+
+    if (!columnType.like(other.columnType)) {
+      mismatches.add(
+          'Type mismatch: Expected "$columnType", found "${other.columnType}".');
+    }
+
+    if (isNullable != other.isNullable) {
+      mismatches.add(
+          'Nullability mismatch: Expected "$isNullable", found "${other.isNullable}".');
+    }
+
+    if (columnDefault != other.columnDefault) {
+      mismatches.add(
+          'Default value mismatch: Expected "$columnDefault", found "${other.columnDefault}".');
+    }
+    return mismatches;
   }
 }
 
 /// Comparison methods for [IndexDefinition].
 extension IndexComparisons on IndexDefinition {
-  /// Returns true if the index structure is identical to the [other]. Ignores
-  /// comparisons of Dart types and names.
-  bool like(
-    IndexDefinition other, {
-    bool ignoreCase = false,
-  }) {
+  /// Compares this index definition with [other], returning a list of mismatches.
+  /// Returns an empty list if the indexes are identical.
+  List<String> like(IndexDefinition other, {bool ignoreCase = false}) {
+    List<String> mismatches = [];
+
     if (ignoreCase) {
-      if (other.indexName.toLowerCase() != indexName.toLowerCase()) {
-        return false;
+      if (indexName.toLowerCase() != other.indexName.toLowerCase()) {
+        mismatches.add(
+            'Index name mismatch: Expected "$indexName", found "${other.indexName}".');
       }
     } else {
-      if (other.indexName != indexName) {
-        return false;
+      if (indexName != other.indexName) {
+        mismatches.add(
+            'Index name mismatch: Expected "$indexName", found "${other.indexName}".');
       }
     }
 
-    return other.isPrimary == isPrimary &&
-        other.isUnique == isUnique &&
-        other.predicate == predicate &&
-        other.tableSpace == tableSpace &&
-        other.type == type;
+    if (type != other.type) {
```

Comment:
Suggestion: Tests are missing for the types below for index.

Reply:
Please validate that all mismatches in the file are covered in some test.

Might be a good idea to create multiple test files that cover differnt areas, this would be covered in an `index` test file.

Reply:
Done!